### PR TITLE
Add property query filtering snippet.

### PR DESCRIPTION
**Pull request by dgay42**

### CODE REVIEW

**Comment by eddavisson** [comment]
@stephenplusplus, can you review this?

File: datastore/concepts.js [hunk 1]
```diff
@@ -619,6 +619,35 @@ Metadata.prototype.testPropertyByKindRunQuery = function(callback) {
   this.datastore.runQuery(query, callback);
 };
 
+Metadata.prototype.testPropertyFilteringRunQuery = function(callback) {
+  datastore.createQuery = this.datastore.createQuery;
+
+  // [START property_filtering_run_query]
+  var query = datastore.createQuery('__property__')
+    .select('__key__')
+    .filter('__key__ >=', datastore.key(['__kind__', 'Task', '__property__', 'priority']));
```
Comment (stephenplusplus) [inline datastore/concepts.js]: I would consider pulling this key out into a variable, just since it's getting pretty long.

**Comment by stephenplusplus** [comment]
Thanks for pinging me. This LGTM, it just needs a test in `test/datastore/metadata.test.js`.

Seeing this reminds me gcloud-node has made some improvements around Key objects, making it easier to reference a name/id, kind, parent, etc. I'll make an issue to remind myself to send the update here.

File: datastore/concepts.js (hunk 1 above)
Comment (dgay42) [inline datastore/concepts.js]: Done.

**Comment by dgay42** [comment]
Added test to metadata.test.js
(where do I find instructions on running the tests?)

**Comment by stephenplusplus** [comment]
You can run `npm test`, defined here: https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/master/package.json#L34

**Comment by dgay42** [comment]
I'll need more instructions than just running "npm test" to actually successfully run the tests.

So far:
- installed lots of stuff
- fixed .jshintrc
- set TEST_PROJECT_ID to a random App Engine app I own

That got it up to:
    1) express-memcached-session should return 200 and Hello World
    ✓ geddy: dependencies should install (336ms)
stderr: 

stderr: events.js:72

stderr:         throw er; // Unhandled 'error' event
              
stderr: ^

stderr: Error: bind EADDRINUSE
    at errnoException (net.js:901:11)
    at net.js:1081:30
    at Object.1:1 (cluster.js:592:5)
    at handleResponse (cluster.js:171:41)
    at respond (cluster.js:192:5)
    at handleMessage (cluster.js:202:5)
    at process.EventEmitter.emit (events.js:117:20)
    at handleMessage (child_process.js:318:10)
    at child_process.js:392:7
    at process.handleConversion.net.Native.got (child_process.js:91:7)

and then it started spewing errors...

**Comment by stephenplusplus** [comment]
// @jmdobry

**Comment by jmdobry** [comment]
Based on the `Error: bind EADDRINUSE` error, it looks like port 8080 was in use before you ran the tests. When the tests run, each app engine sample starts up one at a time, each getting tested. They all will try to bind to port 8080.

I need to improve the testing instructions.

**Comment by pcostell** [comment]
I'm able to reproduce David's issue consistently. After existing I don't see any processes connected to port 8080. It seems like one of the tests doesn't properly release the port. While `npm test` is running, I see an item in netstat appear showing node is holding the port, however on killing the tests that process disappears and releases the port.

**Comment by pcostell** [comment]
Filed #49

File: datastore/concepts.js [hunk 2]
```diff
@@ -619,6 +619,36 @@ Metadata.prototype.testPropertyByKindRunQuery = function(callback) {
   this.datastore.runQuery(query, callback);
 };
 
+Metadata.prototype.testPropertyFilteringRunQuery = function(callback) {
+  datastore.createQuery = this.datastore.createQuery;
+
+  // [START property_filtering_run_query]
+  var startKey = datastore.key(['__kind__', 'Task', '__property__', 'priority'])
```
Comment (pcostell) [inline datastore/concepts.js]: Missing semi-colon (note adding it will make the ltl).

**Comment by pcostell** [comment]
After fixing the semi-colon + ltl, this passes the datastore/concepts/metadata tests. Note there are still some other tests that are failing but I believe these are unrelated.

**Pull request closed without merging**